pkg/monitoring/services/services.go

```diff
@@ -0,0 +1,290 @@
+
+			// most likely impossible but still need to check the boundaries
+			if colIndex >= len(parts) {
+				break
+			}
+
+			// if column is the last - join all the data left in the row because it can contain spaces
+			if colIndex == len(columns)-1 {
+				rowColumnsValues[colName] = strings.Join(parts[colIndex:], " ")
+				break
+			}
+
+			rowColumnsValues[colName] = parts[colIndex]
+		}
+
+		services = append(services,
+			SystemdService{UnitFile: rowColumnsValues["UNIT"], LoadState: rowColumnsValues["LOAD"], ActiveState: rowColumnsValues["ACTIVE"], SubState: rowColumnsValues["SUB"], Description: rowColumnsValues["DESCRIPTION"]},
+		)
+	}
+
+	return services, scanner.Err()
+}
+
+var sysVinitServiceRE = regexp.MustCompile(`^\s+\[\s+([\+\-\?]])\s+\]\s+(.*)$`)
+
+// ListSysVinitServices list SysVinit services via `service --status-all`
+func ListSysVinitServices() ([]SysVService, error) {
+	cmd := exec.Command("service",
+		"--status-all",
+	)
+
+	setPathEnvVar(cmd)
+	var outb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &outb
+	err := cmd.Run()
+	if err != nil {
+		errOutput, _ := ioutil.ReadAll(&outb)
+		return nil, fmt.Errorf("service: %s, %s", err.Error(), string(errOutput))
+	}
+
+	var services []SysVService
+	scanner := bufio.NewScanner(&outb)
+
+	for scanner.Scan() {
+		parts := sysVinitServiceRE.FindStringSubmatch(scanner.Text())
+
+		var state string
+
+		switch parts[1] {
+		case "+":
+			state = "running"
+		case "-":
+			state = "stopped"
+		case "?":
+			state = "unknown"
+		}
+
+		services = append(services,
+			SysVService{UnitFile: parts[2], Status: state},
+		)
+	}
+
+	return services, scanner.Err()
+}
+
+// ListUpstartServices list upstart services via `initctl list`. Returns []SysVService because Upstart is compatible with SysVInit and has the same details
+func ListUpstartServices() ([]SysVService, error) {
+	cmd := exec.Command("initctl",
+		"list",
+	)
+	setPathEnvVar(cmd)
+
+	var outb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &outb
+	err := cmd.Run()
+	if err != nil {
+		errOutput, _ := ioutil.ReadAll(&outb)
+		return nil, fmt.Errorf("Initctl: %s, %s", err.Error(), string(errOutput))
+	}
+
+	var services []SysVService
+	scanner := bufio.NewScanner(&outb)
+	for scanner.Scan() {
+		parts := strings.Fields(scanner.Text())
+
+		if len(parts) < 2 {
+			continue
+		}
+
+		// filter network interfaces from results e.g.
+		// network-interface-security (network-interface/ip6_vti0) start/running
+		// network-interface (dummy0) start/running
+		if strings.HasPrefix(parts[0], "network-interface") {
+			continue
+		}
+
+		stateParts := strings.Split(parts[1], "/")
+
+		services = append(services,
+			SysVService{UnitFile: parts[0], Status: stateParts[1]},
+		)
+	}
+
+	return services, scanner.Err()
+}
+
+func isSystemd() bool {
+	if _, err := os.Stat("/run/systemd/system"); err == nil {
+		return true
+	}
+	return false
+}
+
+func isUpstart() bool {
+	if _, err := os.Stat("/sbin/upstart-udev-bridge"); err == nil {
```

Comment:
Does the alone presence of the file _really_ indicate that upstart is being used?